README
RUN INSTRUCTIONS

To run the game, you must have pygame installed.  The game also requires the
fortune engine installed.  You may find it in the fortune hunter wiki:
https://fedorahosted.org/fortune_hunter/wiki/FortuneEngine

To start the game in terminal run the command "python PPActivity.py"
To start it on an xo, after installing just click the Produce Puzzle Icon which
will run the activity.py file.

GAME PLAY INSTRUCTIONS
To begin, click on the Puzzle Produce icon on your XO.

The game will start immediately.

Choose to start the game or exit. When the game starts you'll have to enter your
guess as to what the second fruit represents. The left shift key brings up your
menu to enter your guess, hitting enter inputs your guess. Check how you've been
doing by watching your score at the bottom right!


NEWS
Version 1:
First release


PPGame.py
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA

# Import section
from fortuneengine.GameEngineElement import GameEngineElement
from Constants import *
from AnswerMenu import AnswerMenuHolder
import pygame
import random

ADDNUM = 1
SUBNUM = 2
MULNUM = 3
DIVNUM = 4

class PPGame(GameEngineElement):

    def __init__(self):
        GameEngineElement.__init__(self, has_draw=True, has_event=True)
        sound = pygame.mixer.music.load(SOUND_PATH+ 'background_music.wav')
        #pygame.mixer.music.play(100, 0)
        self.score = [0,0]
        #[correct, incorrect]
        self.lastCorrect = 2
        #0 = wrong, 1 = right, 2 = skip drawing, 3 = enter a guess!
        self.images = {}
        self.__load_images()
        self.player_input = ''
        self.__setup()

        self.add_to_engine()
        
    def menu_callback(self, selection, menu):
        if selection == 'enter':
            if self.player_input == '':
                self.player_input = '0'
            self.__update()
        elif selection == 'clear':
                self.player_input = ''
        else:
                self.player_input = self.player_input + selection
        
    def __load_images(self):
        apple = pygame.image.load(IMAGE_PATH + "apple.png").convert_alpha()
        orange = pygame.image.load(IMAGE_PATH + "orange.png").convert_alpha()
        banana = pygame.image.load(IMAGE_PATH + "banana.png").convert_alpha()
        plum = pygame.image.load(IMAGE_PATH + "plum.png").convert_alpha()
        pear = pygame.image.load(IMAGE_PATH + "pear.png").convert_alpha()
        mango = pygame.image.load(IMAGE_PATH + "mango.png").convert_alpha()
        strawberry = pygame.image.load(IMAGE_PATH + "strawberry.png").convert_alpha()
        kiwi = pygame.image.load(IMAGE_PATH + "kiwi.png").convert_alpha()
        watermelon = pygame.image.load(IMAGE_PATH + "watermelon.png").convert_alpha()
        mult = pygame.image.load(ICON_PATH + "multiply.png").convert_alpha()
        div = pygame.image.load(ICON_PATH + "divide.png").convert_alpha()
        add = pygame.image.load(ICON_PATH + "plus.png").convert_alpha()
        sub = pygame.image.load(ICON_PATH + "minus.png").convert_alpha()
        equals = pygame.image.load(ICON_PATH + "equals.png").convert_alpha()
        
        self.images["apple"] = apple
        self.images["orange"] = orange
        self.images["banana"] = banana
        self.images["plum"] = plum
        self.images["pear"] = pear
        self.images["mango"] = mango
        self.images["strawberry"] = strawberry
        self.images["kiwi"] = kiwi
        self.images["watermelon"] = watermelon
        self.images[MULNUM] = mult
        self.images[DIVNUM] = div
        self.images[ADDNUM] = add
        self.images[SUBNUM] = sub
        self.images["equals"] = equals
                
    def remove_from_engine(self):
        super( PPGame, self ).remove_from_engine()
        
    def add_to_engine(self):
        super( PPGame, self).add_to_engine()
                    
    # Main game function
    def __setup(self):
        
        #fruit reference dict
        fruit_dictionary = {1:"apple", 2:"orange", 3:"banana", 4:"plum",
                           5:"pear", 6:"mango", 7:"strawberry", 8:"kiwi", 9:"watermelon"}
        
        self.fruit_list = []
        self.player_input = '' 
        
        # Declare row and column sum arrays.
        self.indexes = []
        self.Lsum = 0
        self.Rnum = 0
        self.answer = 0
        
        random.seed()
        #index[] 0 and 1 are the operations in the equation
        #index[2] is the fruit appearing again on the right side
        self.indexes.append(random.randint(1,4))
        self.indexes.append(random.randint(1,4))
        self.indexes.append(random.randint(0,1))
        self.fruit_list.append(self.images[fruit_dictionary[random.randint(1,4)]])
        self.fruit_list.append(self.images[fruit_dictionary[random.randint(5,9)]])

        #build the Left Side of the equation
        if self.indexes[0] == ADDNUM:
            self.Lsum = self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[0])) + self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[1]))
        elif self.indexes[0] == SUBNUM:
            self.Lsum = self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[0])) - self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[1]))
        elif self.indexes[0] == MULNUM:
            self.Lsum = self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[0])) * self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[1]))
            if self.Lsum == 0:
                self.__setup()
        elif self.indexes[0] == DIVNUM:
            self.Lsum = self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[0])) / self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[1]))
            if self.Lsum == 0:
                self.__setup()
        
        self.answer = self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[~(self.indexes[2]) & 1]))
        
        #Build the right unknown
        if self.indexes[1] == ADDNUM:
            self.Rnum = self.Lsum - self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[self.indexes[2]]))
        elif self.indexes[1] == SUBNUM:
            self.Rnum = self.Lsum + self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[self.indexes[2]]))
        elif self.indexes[1] == MULNUM:
            self.Rnum = self.Lsum / self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[self.indexes[2]]))
        elif self.indexes[1] == DIVNUM:
            self.Rnum = self.Lsum * self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[self.indexes[2]]))

    def reverse_lookup(self, d, v):
        for k in d:
            if d[k] == v:
                return k
        raise ValueError
        
    def __check(self):
        if(self.player_input == ''):
            self.lastCorrect = 3
        elif(int(self.player_input) == self.answer):
            self.score[0] += 1
            self.lastCorrect = 1
            self.__setup()
        else:
            self.score[1] += 1
            self.lastCorrect = 0
            self.lastAnswer = self.answer
            self.__setup()
        
        
    def __update(self):
        self.game_engine.get_object('answermenu').remove_from_engine()
        
    def event_handler(self, event):
        """Update the menu and get input for the menu."""
        return_val = False
        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_RETURN:
                return_val = True
                self.__check()
                self.game_engine.set_dirty()
            if event.key == (pygame.K_LSHIFT or pygame.K_RSHIFT):
                return_val = True
                self.game_engine.set_dirty()
                self.game_engine.add_object('answermenu', AnswerMenuHolder(self.menu_callback))
                self.game_engine.get_object('answermenu').show_menu('enter')
    
    def draw(self,screen,time_delta):
        fruit_dictionary = {1:"apple", 2:"orange", 3:"banana", 4:"plum",
                5:"pear", 6:"mango", 7:"strawberry", 8:"kiwi", 9:"watermelon"}
        
        #draw background & setup
        bg = pygame.image.load(IMAGE_PATH + "bg.gif").convert()
        screen.blit(bg,(0,0))
        font = pygame.font.Font(None, 24)
        ren = font.render("Press shift to enter guess. Press enter to check solution!", 1, [255,255,255])
        screen.blit(ren, (20, 630))
        font = pygame.font.Font(None, 64)
        ren = font.render("Solve the Equation!", 1, [0,0,0])
        screen.blit(ren, (275, 120))
        
        
        #First Equation
        x = 250
        y = 220
        for fruit in self.fruit_list:
            screen.blit(fruit, (x,y))
            x += 140
        screen.blit(self.images[self.indexes[0]], (320, y+5))
        screen.blit(self.images["equals"], (x - 45, y+5))
        ren = font.render(str(self.Rnum), 1, [0, 0, 0])
        screen.blit(ren, (x+35,y+5))
        screen.blit(self.images[self.indexes[1]], (x+105, y+5))
        screen.blit(self.fruit_list[self.indexes[2]], (x+175, y))
        
        #second equation
        x=390
        y=320
        screen.blit(self.fruit_list[self.indexes[2]], (x,y))
        screen.blit(self.images["equals"], (x+70,y+5))
        ren = font.render(str(self.reverse_lookup(fruit_dictionary, self.reverse_lookup(self.images, self.fruit_list[self.indexes[2]]))), 1, [0,0,0])
        screen.blit(ren, (x+140,y))
        
        #third equation
        x=390
        y=420
        #nand to get opposite fruit
        screen.blit(self.fruit_list[~(self.indexes[2]) & 1], (x,y))
        screen.blit(self.images["equals"], (x+70,y+5))
        if self.player_input == '':
            ren = font.render("?",1,[0,0,0])
            screen.blit(ren,(x+140,y))
        else:
            ren = font.render(self.player_input,1,[0,0,0])
            screen.blit(ren,(x+140,y))
            
        #draw score and last guess in/correct
        font = pygame.font.Font(None, 24)
        if(self.lastCorrect == 0): 
            ren = font.render("Almost! The answer was " + str(self.lastAnswer), 1, [255,255,255])
        elif(self.lastCorrect == 1):
            ren = font.render("Good job!", 1, [255,255,255])
        elif(self.lastCorrect == 3):
            ren = font.render("Oops, you forgot to enter an answer.", 1, [255,255,255])
        
        if(self.lastCorrect != 2):
            screen.blit(ren, (20, 602))
            
        ren = font.render("#Correct = " + str(self.score[0]) + " and #Incorrect = " + str(self.score[1]), 1, [255,255,255])
        screen.blit(ren, (670,630))
            
        



activity.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Copyright 2007-8 One Laptop per Child Association, Inc.
# Written by C. Scott Ananian <cscott@laptop.org>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
"""Activity helper classes."""
from sugar.activity import activity

# Set to false to hide terminal and auto quit on exit
DEBUG_TERMINAL = False

class ProducePuzzle(activity.Activity):
    """Activity subclass built around the Vte terminal widget."""
    def __init__(self, handle):
        import gtk, pango, platform, sys
        from ctypes import cdll

        if platform.machine().startswith('arm'):
            pass # FIXME
        else:
            if platform.architecture()[0] == '64bit':
                vte_path = "x86-64"
            else:
                vte_path = "x86"
            vte = cdll.LoadLibrary("lib/%s/libvte.so.9" % vte_path)
        sys.path.append("lib/%s" % vte_path)

        import vte

        super(ProducePuzzle, self).__init__(handle, create_jobject=False)
        self.__source_object_id = None

        # creates vte widget
        self._vte = vte.Terminal()

        if DEBUG_TERMINAL:
            toolbox = activity.ActivityToolbox(self)
            toolbar = toolbox.get_activity_toolbar()
            self.set_toolbox(toolbox)

            self._vte.set_size(30,5)
            self._vte.set_size_request(200, 300)
            font = 'Monospace 10'
            self._vte.set_font(pango.FontDescription(font))
            self._vte.set_colors(gtk.gdk.color_parse ('#E7E7E7'),
                                 gtk.gdk.color_parse ('#000000'),
                                 [])

            vtebox = gtk.HBox()
            vtebox.pack_start(self._vte)
            vtesb = gtk.VScrollbar(self._vte.get_adjustment())
            vtesb.show()
            vtebox.pack_start(vtesb, False, False, 0)
            self.set_canvas(vtebox)

            toolbox.show()
            self.show_all()
            toolbar.share.hide()
            toolbar.keep.hide()

        # now start subprocess.
        self._vte.connect('child-exited', self.on_child_exit)
        self._vte.grab_focus()
        bundle_path = activity.get_bundle_path()
        self._pid = self._vte.fork_command \
            (command='/bin/sh',
             argv=['/bin/sh','-c',
             'python %s/PPActivity.py' % bundle_path],
             envv=["PYTHONPATH=%s/library" % bundle_path],
             directory=bundle_path)
    def on_child_exit(self, widget):
        """This method is invoked when the user's script exits."""
        if not DEBUG_TERMINAL:
            import sys
            sys.exit()



Constants.py
# -*- coding: utf-8 -*-
#
#       constants.py
#       
#       Copyright 2010 Kevin Hockey <blitz@blitzkev>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.

import os.path
from gettext import gettext as _

BASE_PATH = os.path.dirname(__file__)
SOUND_PATH = BASE_PATH + "/sounds/"
IMAGE_PATH = BASE_PATH + "/graphics/"
ICON_PATH = BASE_PATH + "/icons/"



AnswerMenu.py
import pygame
from fortuneengine.GameEngineElement import GameEngineElement
from Constants import IMAGE_PATH
from gettext import gettext as _

NORMAL_MENU = 1
GRID_MENU = 3

class AnswerMenuHolder( GameEngineElement ):
    def __init__(self, callback):
        GameEngineElement.__init__(self, has_draw=False, has_event=False)
        self.menu = None
        self.callback = callback
        self.font = pygame.font.SysFont("cmr10",26,False,False)
        self.add_to_engine()

    def remove_from_engine(self):
        super( AnswerMenuHolder, self ).remove_from_engine()
        self.clear_menu()

    def menu_called(self, id):
        self.callback(id, self)

    def clear_menu(self):
        if self.menu:
            self.menu.remove_from_engine()
            self.menu = None

    def show_menu(self,id):
        if self.is_in_engine():
            self.clear_menu()
        else:
            self.add_to_engine()
            
        if id == "enter":
            menu_type = GRID_MENU
            menu_options = [
                        ['1', lambda: self.menu_called('1')],
                        ['2', lambda: self.menu_called('2')],
                        ['3', lambda: self.menu_called('3')],
                        ['4', lambda: self.menu_called('4')],
                        ['5', lambda: self.menu_called('5')],
                        ['6', lambda: self.menu_called('6')],
                        ['7', lambda: self.menu_called('7')],
                        ['8', lambda: self.menu_called('8')],
                        ['9', lambda: self.menu_called('9')],
                        [_("C"), lambda: self.menu_called('clear')],
                        ['0', lambda: self.menu_called('0')],
                        [_("E"), lambda: self.menu_called('enter')],
            ]
        else:
            print "Invalid Menu", id
            return
        self.menu = AnswerMenu(menu_options)
        
class AnswerMenu(GameEngineElement):
    def __init__(self, options):
        GameEngineElement.__init__(self, has_draw=True, has_event=True)

        self.menu = Menu(options)

        self.menu.set_pos(650, 420)
        self.add_to_engine()

    def event_handler(self, event):
        self.game_engine.set_dirty()
        return self.menu.update(event)

    def draw(self,screen,time_delta):
        self.menu.draw( screen )

class Menu(object):
    def __init__(self, options):
        """Initialize the EzMenu! options should be a sequence of lists in the
        format of [option_name, option_function]"""

        self.options = options
        self.x = 0
        self.y = 0
        self.cols = 3
        self.font = pygame.font.SysFont("cmr10",24,False,False)
        self.option = 0
        self.width = 1
        self.color = [0, 0, 0]
        self.hcolor = [255, 0, 0]
        self.height = len(self.options)*self.font.get_height()

    def draw(self, surface):
        """Draw the menu to the surface."""
        i=0 # Row Spacing
        h=0 # Selection Spacing
        j=0 # Col Spacing
        xoff = 198/3 - 1
        yoff = 248/4 - 1
        for o in self.options:
            if h==self.option:
                clr = self.hcolor
            else:
                clr = self.color
                
            text = o[0]
            ren = self.font.render(text, 1, clr)
            if ren.get_width() > self.width:
                self.width = ren.get_width()

            newX = self.x + xoff * j
            newY = self.y+ i * yoff            
            pygame.draw.rect(surface, (0, 0, 255), ( newX, newY, 62, 58))
            pygame.draw.rect(surface, (4, 119, 152), ( newX + 2, newY + 2, 58, 54))
            surface.blit(ren, (newX + 26, newY + 26))

            j+=1
            h+=1
            if j >= self.cols:
                i+=1
                j=0

    def update(self, event):
        """Update the menu and get input for the menu."""
        return_val = False
        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_DOWN:
                if self.cols != 1:
                    self.option += self.cols
                else:
                    self.option += 1
                return_val = True
            elif event.key == pygame.K_UP:
                if self.cols != 1:
                    self.option -= self.cols
                else:
                    self.option -= 1
                return_val = True
            elif event.key == pygame.K_RIGHT:
                if self.cols != 1:
                    self.option += 1
                    return_val = True
            elif event.key == pygame.K_LEFT:
                if self.cols != 1:
                    self.option -= 1
                    return_val = True
            elif event.key == pygame.K_RETURN:
                self.options[self.option][1]()
                return_val = True

            self.option = self.option % len(self.options)
        return return_val

    def set_pos(self, x, y):
        """Set the topleft of the menu at x,y"""
        self.x = x + 3
        self.y = y + 51


PPMainMenu.py
# -*- coding: utf-8 -*-
#
#       PPMainMenu.py
#       
#       Copyright 2010 Kevin Hockey <blitz@blitzkev>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.

import pygame, ezmenu
from fortuneengine.GameEngineElement import GameEngineElement

class MainMenuHolder( GameEngineElement ):
    def __init__(self, callback, background=None, width=952, height=732):
        GameEngineElement.__init__(self, has_draw=True, has_event=False)
        self.menu = None
        self.callback = callback
        bg = pygame.image.load(background).convert()
        self.background = pygame.transform.scale(bg, (width, height))
        self.width = width
        self.height = height

    def remove_from_engine(self):
        super( MainMenuHolder, self ).remove_from_engine()
        self.clear_menu()

    def draw(self,screen,time_delta):
        if self.background:
            screen.blit(self.background,(0,0))
        else:
            screen.fill((0, 0, 255))

    def menu_called(self, id):
        self.callback(id, self)

    def clear_menu(self):
        if self.menu:
            self.menu.remove_from_engine()
            self.menu = None

    def show_menu(self,id):
        if self.is_in_engine():
            self.clear_menu()
        else:
            self.add_to_engine()

        if id == "title":
            menu_options = [
                        ["Produce Puzzle", lambda: self.show_menu("adventure"), "Begin a new game!"],
                        ["Exit Game", lambda: self.game_engine.stop_event_loop(), "Exit the game."]
            ]

        elif id == "adventure":
            menu_options = [
                        ["New Game",  lambda: self.menu_called("new"), "Start a new game."],
                        ["Return to Title", lambda: self.show_menu("title"), "Return to the title menu."]
            ]
        else:
            print "Invalid Menu", id
            return
        self.menu = MainMenu(menu_options, self.width, self.height)

class MainMenu(GameEngineElement):
    def __init__(self, game_menu, width=800, height=400):
        GameEngineElement.__init__(self, has_draw=True, has_event=True)
        self.menu = ezmenu.EzMenu(game_menu)
        self.menu.center_at(width - (width/3), height/2)
        self.menu.help_text_at( 0, height-(height/8))
        self.menu.set_font(pygame.font.SysFont("Arial", 20))
        self.menu.set_highlight_color((0, 255, 0))
        self.menu.set_normal_color((255, 255, 255))
        self.add_to_engine()

    def event_handler(self, event):
        self.game_engine.set_dirty()
        return self.menu.update(event)

    def draw(self,screen,time_delta):
        self.menu.draw( screen )


PPGameManager.py
# -*- coding: utf-8 -*-
#
#       PPGameManager.py
#       
#       Copyright 2010 Kevin Hockey <blitz@blitzkev>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.
import pygame
from fortuneengine.GameEngineElement import GameEngineElement
from PPGame import PPGame


class PPGameManager(GameEngineElement):
    def __init__(self):
        GameEngineElement.__init__(self, has_draw=False, has_event=True)
        self.add_to_engine()

        game_size_ratio_x = self.game_engine.width/1200.0
        game_size_ratio_y = self.game_engine.height/900.0

        term_width_offset = game_size_ratio_x * 200
        term_height = game_size_ratio_y * 200
        term_height_offset = game_size_ratio_y * 700
        term_width = game_size_ratio_x * 1000



        #self.game_engine.add_object('game', PPGameHolder())

    def event_handler(self, event):
        if event.type == pygame.KEYDOWN:
            newKey=pygame.key.name(event.key)
            if newKey=='escape':
                self.game_engine.stop_event_loop()
                return True




PPActivity.py
# -*- coding: utf-8 -*-
#
#       PPActivity.py
#       
#       Copyright 2010 Kevin Hockey <blitzkev@gmail.com>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.

from fortuneengine.GameEngine import GameEngine
from PPGameManager import PPGameManager
from PPMainMenu import MainMenuHolder
from PPGame import PPGame
from Constants import IMAGE_PATH

ge = GameEngine(width=952, height=714, always_draw=False)


def start_game():
    ge.add_object('manager', PPGameManager() )

def menu_screen():
    ge.add_object('menu', MainMenuHolder( menu_called, IMAGE_PATH + "bg.gif", width=ge.width, height=ge.height))
    ge.get_object('menu').show_menu('title')

def menu_called(id, menu):
    if id == 'new':
        menu.remove_from_engine()
        start_game()
        ge.remove_object('menu')
        ge.add_object('game', PPGame() )
    else:
        print "MENU CALLED %s" % id

# Build menu and add to engine.  Then show menu
menu_screen()

# Start event loop
ge.start_main_loop()
